### PR TITLE
Actuators message for JointPositionController.

### PR DESCRIPTION
**Pull request by bperseghetti**
# 🎉 New feature

Adds ability to control the position of joints with the position field of the actuators message.

Closes: #1974 

As seen in the Gazebo community meeting:
https://vimeo.com/812911652#t=39m58s

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [x] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by bperseghetti** [comment]
Still needs the tests added.

**Comment by bperseghetti** [comment]
@azeey @mjcarroll good to merge now?

File: src/systems/joint_position_controller/JointPositionController.cc [hunk 1]
```diff
@@ -190,9 +202,26 @@ void JointPositionController::Configure(const Entity &_entity,
...
+        _sdf->Get<int>("actuatorNumber");
+      this->dataPtr->useActuatorMsg = true;
+    }
+    else
+    {
+      gzerr << "Please specify an actuatorNumber" <<
+        "to use Actuator position message control." << std::endl;
+    }
+  }
+
```
Comment (mjcarroll) [inline src/systems/joint_position_controller/JointPositionController.cc:220]: I don't know if I caught this in the other PR, but we do underscores in sdf tags:

```suggestion
  if (_sdf->HasElement("use_actuator_msg") &&
    _sdf->Get<bool>("use_actuator_msg"))
  {
    if (_sdf->HasElement("actuator_number"))
    {
      this->dataPtr->actuatorNumber =
        _sdf->Get<int>("actuator_number");
      this->dataPtr->useActuatorMsg = true;
    }
    else
    {
      gzerr << "Please specify an actuator_number" <<
        "to use Actuator position message control." << std::endl;
    }
  }

```

File: test/worlds/joint_position_controller_actuators.sdf [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        <axis>
+          <xyz>0 0 1</xyz>
+        </axis>
+      </joint>
+      <plugin
+        filename="gz-sim-joint-position-controller-system"
+        name="gz::sim::systems::JointPositionController">
+        <joint_name>j1</joint_name>
+        <use_actuator_msg>true</use_actuator_msg>
+        <actuatorNumber>0</actuatorNumber>
```
Comment (mjcarroll) [inline test/worlds/joint_position_controller_actuators.sdf:119]:
```suggestion
        <actuator_number>0</actuator_number>
```

File: src/systems/joint_position_controller/JointPositionController.cc (hunk 1 above)
Comment (bperseghetti) [reply src/systems/joint_position_controller/JointPositionController.cc:220]: @mjcarroll Ok, I can update all of them easily, I was following the style that I had seen from:

https://github.com/gazebosim/gz-sim/blob/14d583283a860efa71d80d9d6d620ba0b31afa97/src/systems/multicopter_motor_model/MulticopterMotorModel.cc#L286-L290

Should I change it in multicopter_motor_model to match as well?

Comment (bperseghetti) [reply src/systems/joint_position_controller/JointPositionController.cc:220]: Here is my fix for the previous ones: https://github.com/gazebosim/gz-sim/pull/1976

**Comment by bperseghetti** [comment]
@mjcarroll I [updated what you requested](https://github.com/gazebosim/gz-sim/pull/1954/commits/b7671af959743aef99bbff667ed7bbec951cd4a7).

I also [force pushed](https://github.com/gazebosim/gz-sim/commit/5d87ee2fb1156ee76436359f1adf9e95c3c3ad16) the same changes to JC PR #1953

As well as [fixed Ackermann and unified MulticopterMotorModel](https://github.com/gazebosim/gz-sim/pull/1976) with a deprecation warning and ability to use same `<actuator_number>`.

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**